src/lib/client.ts

```diff
@@ -2106,7 +2106,7 @@ export default class MqttClient extends TypedEventEmitter<MqttClientEventCallbac
 		if (
 			!this._firstConnection &&
 			(this.options.clean ||
-				(this.options.protocolVersion === 5 &&
```

Comment:
Could you add a comment here with the links to MQTT 3.1.1 specs that explain this?

Reply:
Done.